---
layout: issue
title: "Default ErrorController created by Zend_Tool generates notices."
id: ZF-10071
---

ZF-10071: Default ErrorController created by Zend\_Tool generates notices.
--------------------------------------------------------------------------

 Issue Type: Bug Created: 2010-06-28T22:35:23.000+0000 Last Updated: 2010-08-25T19:42:03.000+0000 Status: Resolved Fix version(s): - 1.10.8 (25/Aug/10)
 
 Reporter:  Henrique Moody (henriquemoody)  Assignee:  Ralph Schindler (ralph)  Tags: - Zend\_Tool
 
 Related issues: - [ZF-10252](/issues/browse/ZF-10252)
 
 Attachments: - [Zend\_Tool\_Project\_Context\_Zf\_ControllerFile.diff](/issues/secure/attachment/13178/Zend_Tool_Project_Context_Zf_ControllerFile.diff)
 
### Description

When I create a project with Zend\_Tool, for exemple:

$ cd /var/www $ zf create project test

And after, in the browser I call: "http://localhost/test/public/error/error", I see 3 notices in the development enviroment:

Notice: Trying to get property of non-object in /var/www/test/application/controllers/ErrorController.php on line 10 Notice: Trying to get property of non-object in /var/www/test/application/controllers/ErrorController.php on line 33 Notice: Trying to get property of non-object in /var/www/test/application/controllers/ErrorController.php on line 36

This occurred becouse $this->\_getParam('error\_handler') is null when I am in "/error/error".

Before proceeding this line, must have a verification if $this->\_getParam('error\_handler') is instance of ArrayObject.

 

 

### Comments

Posted by Ramon Henrique Ornelas (ramon) on 2010-08-21T13:14:58.000+0000

In a issue similar the this was fixed ZF-10252 with [r22872](http://framework.zend.com/code/changelog/Standard_Library?cs=22872).

 

 

Posted by Henrique Moody (henriquemoody) on 2010-08-22T10:49:32.000+0000

This does not solve the problem because the parameter "error\_handler" can be passed through the URL:

/error/error?error\_handler=123

You must see my diff, it checks if "error\_handler" is instance of ArrayObject.

 

 

Posted by Henrique Moody (henriquemoody) on 2010-08-25T15:21:34.000+0000

In the 1.10.8 release it was solved.

Thanks!

 

 